src/_P096_eInk.ino

```diff
@@ -0,0 +1,603 @@
+            }           
+            else if(subcommand.equalsIgnoreCase(F("rr")) && argCount == 6)
+            {
+              eInkScreen->drawRoundRect(sParams[0].toInt(), sParams[1].toInt(), sParams[2].toInt(), sParams[3].toInt(), sParams[4].toInt(), Plugin_096_ParseColor(sParams[5]));
+            }          
+            else if(subcommand.equalsIgnoreCase(F("rrf")) && argCount == 7)
+            {
+              eInkScreen->fillRoundRect(sParams[0].toInt(), sParams[1].toInt(), sParams[2].toInt(), sParams[3].toInt(), sParams[4].toInt(), Plugin_096_ParseColor(sParams[6]));
+              eInkScreen->drawRoundRect(sParams[0].toInt(), sParams[1].toInt(), sParams[2].toInt(), sParams[3].toInt(), sParams[4].toInt(), Plugin_096_ParseColor(sParams[5]));
+            } 
+            else if(subcommand.equalsIgnoreCase(F("px")) && argCount == 3)
+            {
+              eInkScreen->drawPixel(sParams[0].toInt(), sParams[1].toInt(), Plugin_096_ParseColor(sParams[2]));
+            } 
+            else 
+            {
+              success = false;
+            }
+          }
+                    
+        }
+        else
+        {
+          //invalid arguments
+          success = false;
+        }
+
+
+        if(!success)
+        {
+          if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+            addLog(LOG_LEVEL_INFO, F("Fail to parse command correctly; please check API documentation"));
+            String log2  = F("Parsed command = \"");
+            log2 += string;
+            log2 += "\"";
+            addLog(LOG_LEVEL_INFO, log2);
+          }
+        }
+        else
+        {
+          if(plugin_096_sequence_in_progress == false)
+            eInkScreen->display();
+        }
+
+        String log;
+        log.reserve(110);                           // Prevent re-allocation
+        log = F("P096-eInk : WRITE = ");
+        log += tmpString;
+        SendStatus(event->Source, log);             // Reply (echo) to sender. This will print message on browser.  
+              
+        break;        
+      }
+  }
+
+
+
+  return success;
+}
+
+
+//Print some text
+//param [in] string : The text to display
+//param [in] X : The left position (X)
+//param [in] Y : The top position (Y)
+//param [in] textSize : The text size (default 1)
+//param [in] color : The fore color (default ILI9341_WHITE)
+//param [in] bkcolor : The background color (default ILI9341_BLACK)
+void Plugin_096_printText(const char *string, int X, int Y, unsigned int textSize, unsigned short color, unsigned short bkcolor)
+{
+  eInkScreen->clearBuffer();
+  eInkScreen->clearDisplay();
+  eInkScreen->setCursor(X, Y);
+  eInkScreen->setTextColor(color, bkcolor);
+  eInkScreen->setTextSize(textSize);
+  String stringOne  = string;
+  String stringTwo = "";
+  stringTwo += (char)247;
+  stringOne.replace("°", stringTwo);
```

Comment:
Do not use extended ASCII or Unicode in source code (not even in comments)

Reply:
Ok I've done this, to allow displaying the "degree" symbol.
The degree symbol is not in standard font and cannot be displayed (print("°") display strange cars)
As my personal main usage is for domotic software, degrees is a very important car.
Is there any other way to do it ?

Reply:
I do have similar replacements in the file StringConverter.ino.
See:

```c++
  {
    // Degree
    const char degree[3]   = { 0xc2, 0xb0, 0 };       // Unicode degree symbol
    const char degreeC[4]  = { 0xe2, 0x84, 0x83, 0 }; // Unicode degreeC symbol
    const char degree_C[4] = { 0xc2, 0xb0, 'C', 0 };  // Unicode degree symbol + captial C
    repl(F("{D}"),   degree,   s, useURLencode);
    repl(F("&deg;"), degree,   s, useURLencode);
    repl(degreeC,    degree_C, s, useURLencode);
  }
```

Reply:
OK thanks.
Done.